### ISSUE TITLE
Simple DEX using Concordium ID Tutorial (Medium article) - Reward $2k (in $CCDs) [+ Video Tutorial Bonus] 

### ISSUE THREAD

**Issue by bogacyigitbasi** [Concordium/concordium-bounty-program#4]
- Implement a basic DEX that allows the trade of CIS-2 tokens including smart contract, front-end, and documentation.
- Tutorial documentation must show all details of the implementation with brief descriptions. 
- Check the user is older than 18 before initiating any operation.
- Open-sourced Github repo
- Hosted on testnet and vercel app or GitHub pages
- If maintenance is required, the participant is responsible for doing it for 3 months.

**Comment by bogacyigitbasi** [comment]
Hey @KizitoNaanma since it is a bounty for a tutorial, I do not want to limit your imagination but ideally:

- the platform should check your age while you are connecting to it which requires running a verifier backend and small front-end work
- Users should be able to create pools by adding any (CIS-2, CIS-2) and (CCD, CIS-2) pairs
- Trading using pools (with slippage logic) needs to be explained & implemented 
- Current status of pools (liquidity, TVL, volume etc.)
- You should use Concordium tools like VSCode extension, ccd-js-gen, etc)
- Since the focus is on the beginner level of developers, focus on simplicity and document it well.

**Comment by AyyanNiazi** [comment]
@bogacyigitbasi i'd like to start work on this

**Comment by bogacyigitbasi** [comment]
Hey @AyyanNiazi go ahead! 3 months is way too long to wait and unfair for the others! Typically, when you have something ready you share ping me. Please let me know if you need anything else.

**Comment by AyyanNiazi** [comment]
@bogacyigitbasi I'll read doc and will try to complete, in case something come I'll definitely will let you know

**Comment by AyyanNiazi** [comment]
Snart contract done, please have a look @bogacyigitbasi 
https://github.com/AyyanNiazi/cocncordium-dex

**Comment by bogacyigitbasi** [comment]
Hey @AyyanNiazi thats great you are super fast, but please note that this is a tutorial. We would like to see comments, explanations, readmes and even visuals if applicable. When all is ready, including the medium article, working demo with UI (a recording perhaps) we can review it. As of now, its a super impressive considering the speed you have but we need more to review it. Great hacks!

**Comment by AyyanNiazi** [comment]
@bogacyigitbasi yes I'm working on it, and creating frontend and backend for it, and then I'll add proper docs, readme and medium article

**Comment by bogacyigitbasi** [comment]
@AyyanNiazi do you need anything else?

**Comment by AyyanNiazi** [comment]
Hi @bogacyigitbasi, I've pushed server code in to the repo now working on the client side where users can provide liquidity and swap tokens

**Comment by bogacyigitbasi** [comment]
Hey @AyyanNiazi  Since it has been two months, I wanted to let you know that we are considering closing this task and potentially include in the next season.

**Comment by binbash0** [comment]
@bogacyigitbasi I want to start working on this project, please don't close.

**Comment by bogacyigitbasi** [comment]
This is closed, wait for the next season @binbash0! DevRel team is cooking something 🥰
